compiler/optimizations/bulkCopyRecords.cpp

```diff
@@ -135,11 +44,25 @@ static bool isAssignment(FnSymbol* fn)
 }
 
 
-static bool isTrivialAssignment(FnSymbol* fn, FnSet& visited)
+static bool isTrivialAssignment(FnSymbol* fn)
 {
   if (! isAssignment(fn))
     return false;
-  if (isTrivial(fn, visited))
+  
+  // The base argument types must match.
+  ArgSymbol* lhs = fn->getFormal(1);
+  ArgSymbol* rhs = fn->getFormal(2);
+  if (lhs->type->getValType() != rhs->type->getValType())
+    return false;
+
+  // Don't consider assignment for class types to be trivial.
+  // That is because a class type represents a pointer, so
+  // we don't really track whether or not it is POD.
```

Comment:
Actually, there is a test that overrides = for a class type.
So, =  *normally* copies class pointers, but apparently it can do something else.
That is why I made that change. Should I include the test in a comment?